### PR TITLE
Fix using rules_kotlin from head

### PR DESCRIPTION
**Pull request by restingbull**
Update local development instructions
Move release_archive to source directory
Add `kotlinc_version` helper for choosing compiler versions.

Fixes Issue #570 

Signed-off-by: restingbull@cabaretmechanique.com <restingbull@cabaretmechanique.com>

### CODE REVIEW

**Comment by cgruber** [comment]
Wait, this is burying the lede, isn't it? are you enabling 1.5 with this commit?

**Comment by restingbull** [comment]
> Wait, this is burying the lede, isn't it? are you enabling 1.5 with this commit?

Shoot, no. Lemme go fix that.

**Comment by jeffzoch** [comment]
@cgruber @restingbull what are the issues w/ 1.5 we are still looking to fix?

**Comment by restingbull** [comment]
> @cgruber @restingbull what are the issues w/ 1.5 we are still looking to fix?

I'm doing 1.5 in another pr.

File: README.md [hunk 1]
```diff
@@ -162,19 +162,14 @@ To choose a different `kotlinc` distribution (1.3 and 1.4 variants supported), d
...
 
-KOTLINC_RELEASE = {
-    "urls": [
-        "https://github.com/JetBrains/kotlin/releases/download/v{v}/kotlin-compiler-{v}.zip".format(v = KOTLIN_VERSION),
-    ],
-    "sha256": KOTLINC_RELEASE_SHA,
-}
-
-kotlin_repositories(compiler_release = KOTLINC_RELEASE)
+kotlin_repositories(
```
Comment (cgruber) [inline README.md:167]: Ugh, thank you for this. I should have done this kotlinc_version thing a long time ago.

**Review by cgruber — APPROVED** [review]
LGTM

**Pull request merged**